splash/network_manager.py

```diff
@@ -372,14 +370,23 @@ def run_response_middlewares(self):
         reply.metaDataChanged.disconnect(self.run_response_middlewares)
         render_options = self._get_render_options(reply.request())
         if render_options:
-            for middleware in self.response_middlewares:
-                middleware.process(reply, render_options)
+            try:
+                for middleware in self.response_middlewares:
+                    middleware.process(reply, render_options)
+            except:
+                self.log("internal error in response middleware", min_level=1)
+                self.log(traceback.format_exc(), min_level=1, format_msg=False)
 
     def createRequest(self, operation, request, outgoingData=None):
```

Comment:
perhaps we should add docstring saying that you cannot return anything else but QNetworkReply from here, if anything else is returned Splash will always crash. I tested this with returning None and it died. It is easy to forget about this if you're used to Python. Usually there is no harm in doing return None somewhere but here it is very harmfful.